k8s-service.yaml

```diff
@@ -0,0 +1,39 @@
+kind: Service
+apiVersion: v1
+metadata:
+  # Unique key of the Service instance
+  name: backend-service
+spec:
+  ports:
+    # Accept traffic sent to port 80
+    - name: http
+      port: 8080
+      nodePort: 30080
+      # Please keep nodeport for running in local k8s cluster as minikube. Needs to be changed to Loadbalancer for cloud driven K8s clusters
+
+  selector:
+    # Loadbalance traffic across Pods matching
+    # this label selector
+    app: backend-service
+  # Create an HA proxy in the cloud provider
+  # with an External IP address - *Only supported
+  # by some cloud providers*
+  type: NodePort
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: redis
+  name: redis
+  # This label matches with the application.yml value of redis application. Any other service
+  # name would not make the end to end testing successful
+spec:
+  ports:
+    - port: 6379
+      nodePort: 30090
+  selector:
+    app: redis-server
+  type: NodePort
```

Comment:
This is redis service, by having redis as a NodePort service, you're exposing it externally. I think it should be ClusterIp.